src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -94,183 +95,125 @@ struct npc_corastrasza : public ScriptedAI
     }
 };
 
-enum NesingwaryTrapper
-{
-    NPC_NESINGWARY_TRAPPER = 25835,
-
-    GO_HIGH_QUALITY_FUR = 187983,
-
-    GO_CARIBOU_TRAP_1   = 187982,
-    GO_CARIBOU_TRAP_2   = 187995,
-    GO_CARIBOU_TRAP_3   = 187996,
-    GO_CARIBOU_TRAP_4   = 187997,
-    GO_CARIBOU_TRAP_5   = 187998,
-    GO_CARIBOU_TRAP_6   = 187999,
-    GO_CARIBOU_TRAP_7   = 188000,
-    GO_CARIBOU_TRAP_8   = 188001,
-    GO_CARIBOU_TRAP_9   = 188002,
-    GO_CARIBOU_TRAP_10  = 188003,
-    GO_CARIBOU_TRAP_11  = 188004,
-    GO_CARIBOU_TRAP_12  = 188005,
-    GO_CARIBOU_TRAP_13  = 188006,
-    GO_CARIBOU_TRAP_14  = 188007,
-    GO_CARIBOU_TRAP_15  = 188008,
-
-    SPELL_TRAPPED       = 46104,
-
-    // Texts
-    SAY_NESINGWARY_1    = 0
-};
+/*######
+## Quest 11865: Unfit for Death
+######*/
 
-#define CaribouTrapsNum 15
-const uint32 CaribouTraps[CaribouTrapsNum] =
+// Gameobjects 187982,187995,187996,187997,187998,187999,188000,188001,188002,188003,188004,188005,188006,188007,188008: Caribou Trap
+enum CaribouTrap
 {
-    GO_CARIBOU_TRAP_1, GO_CARIBOU_TRAP_2, GO_CARIBOU_TRAP_3, GO_CARIBOU_TRAP_4, GO_CARIBOU_TRAP_5,
-    GO_CARIBOU_TRAP_6, GO_CARIBOU_TRAP_7, GO_CARIBOU_TRAP_8, GO_CARIBOU_TRAP_9, GO_CARIBOU_TRAP_10,
-    GO_CARIBOU_TRAP_11, GO_CARIBOU_TRAP_12, GO_CARIBOU_TRAP_13, GO_CARIBOU_TRAP_14, GO_CARIBOU_TRAP_15,
+    EVENT_FUR_SPAWN        = 1,
+    EVENT_SPAWN_TRAPPER    = 2,
+    EVENT_TRAPPER_MOVE     = 3,
+    EVENT_TRAPPER_TEXT     = 4,
+    EVENT_TRAPPER_LOOT     = 5,
+    EVENT_FUR_DESPAWN      = 6,
+    EVENT_TRAPPER_DIE      = 7,
+    EVENT_DESPAWN_ALL      = 8,
+    GO_HIGH_QUALITY_FUR    = 187983,
+    NPC_NESINGWARY_TRAPPER = 25835,
+    SAY_NESINGWARY_1       = 0,
+    SPELL_PLACE_FAKE_FUR   = 46085,
+    SPELL_TRAPPED          = 46104
 };
 
-// 46085 - Place Fake Fur
-class spell_q11865_place_fake_fur : public SpellScript
+struct go_caribou_trap : public GameObjectAI
 {
-    PrepareSpellScript(spell_q11865_place_fake_fur);
+    go_caribou_trap(GameObject* go) : GameObjectAI(go){}
 
-    bool Load() override
+    void Reset() override
     {
-        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+        _placeFir = false;
+        _goFurGUID.Clear();
+        _playerGUID.Clear();
+        _trapperGUID.Clear();
+        me->SetGoState(GO_STATE_READY);
```

Comment:
since dynamic spawns this is no longer used for respawning. JustAppeared() is the way to go here.

Reply:
No idea what you are talking about. You guys just blurt out things. Type the code you want put there. You expect everyone to know everything and then wonder why people get pissed off.

Reply:
What I'm saying is that this whole code in Reset() can be deleted because since dynamic spawns are a thing, respawning gameobjects spawns brand new entities and a new instance of this script. Meaning that everything you do in Reset() is obsolete.

Reply:
Ok is that what you want?

Reply:
yup. It's basically dead code so let's get rid of it :)

Reply:
Thank you.